pkg/config/config.go

```diff
@@ -208,7 +214,34 @@ func validateConfig(cfg *Config) error {
 	return nil
 }
 
-// GetConfig returns a shared instance of config.
-func GetConfig() *Config {
-	return &defaultConfig
+func ParseConfig(path string) (*Config, error) {
+	if path == "" {
+		path = filepath.Join("configs", "config.json")
+	}
+
+	cfg := &Config{}
+	// DeepCopy the default config into the final.
+	{
+		b, _ := json.Marshal(defaultConfig)
+		if err := json.Unmarshal(b, cfg); err != nil {
+			return nil, errors.Wrap(err, "copy default config")
+		}
+	}
+
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, errors.Wrap(err, "open config file")
+	}
+	dec := json.NewDecoder(f)
+	dec.DisallowUnknownFields()
+	for {
+		// Override defaults with the custom configs.
+		if err := dec.Decode(cfg); err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, errors.Wrap(err, "parse config")
+		}
+
+	}
```

Comment:
Somewhere in here you need to also expand all env variables from the env file

`err = godotenv.Load(cfg.EnvFile)`